plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -1245,8 +1245,12 @@ def main():
                 module.exit_json(changed=False, vmid=vmid, msg="VM with vmid <%s> already exists" % vmid)
             elif proxmox.get_vmid(name, ignore_missing=True) and not (update or clone):
                 module.exit_json(changed=False, vmid=proxmox.get_vmid(name), msg="VM with name <%s> already exists" % name)
-            elif (not node) or (not name):
-                module.fail_json(msg='node, name is mandatory for creating/updating vm')
+            elif not node:
+                module.fail.json(msg='node is mandatory for creating/updating vm')
+            elif update and not any([vmid, name]):
+                module.fail_json(msg='vmid or name is mandatory for updating vm')
+            elif not update and not name:
+                module.fail_json(msg='name is mandatory for creating vm')
```

Comment:
Is it actually required? When using the GUI, the name is automatically set to the ID if empty. Not sure about the API though.

Reply:
only requirements are `node` and `vmid` per the documentation.